rest-api-spec/src/main/resources/rest-api-spec/test/explain/40_mix_typeless_typeful.yml

```diff
@@ -0,0 +1,56 @@
+---
+"Explain with typeless API on an index that has types":
+
+ - skip:
+      version: " - 6.99.99"
+      reason:  Typeless APIs were introduced in 7.0.0
+
+ - do:
+      indices.create: # not using include_type_name: false on purpose
+          index: index
+          body:
+              mappings:
+                  not_doc:
+                      properties:
+                          foo:
+                              type: "keyword"
+
+ - do:
+      index:
+          index:  index
+          type:   not_doc
+          id:     1
+          body:   { foo: bar }
+
+ - do:
+      indices.refresh: {}
+
+ - do:
+      catch:  missing
+      explain:
+          index:  index
+          type:   some_random_type
+          id:     1
+          body:
+            query:
+              match_all: {}
+
+ - match:   { _index:   "index" }
+ - match:   { _type:    "some_random_type" }
+ - match:   { _id:      "1"}
+ - match:   { matched:  false}
+
+ - do:
+      explain:
+          index:  index
+          type:   _doc #todo: make _explain typeless and remove this
+          id:     1
+          body:
+            query:
+              match_all: {}
+
+ - match:   { _index:   "index" }
+ - match:   { _type:    "_doc" }
+ - match:   { _id:      "1"}
+ - is_true: matched
+ - match:   { explanation.value: 1 }
```

Comment:
do we also need to check whether the doc is found when specifying not_doc as a type?

Reply:
we have one already